src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestTargeting.ll

```diff
@@ -0,0 +1,19 @@
+define void @Microsoft__Quantum__Testing__QIR__NoArgs__body() {
+entry:
+  %q = call %Qubit* @__quantum__rt__qubit_allocate()
+  %0 = call %Result* @__quantum__qis__mz(%Qubit* %q)
+  %1 = call %Tuple* @__quantum__rt__tuple_create(i64 mul nuw (i64 ptrtoint (i1** getelementptr (i1*, i1** null, i32 1) to i64), i64 2))
+  %2 = bitcast %Tuple* %1 to { %Callable*, %Qubit* }*
+  %3 = getelementptr inbounds { %Callable*, %Qubit* }, { %Callable*, %Qubit* }* %2, i32 0, i32 0
+  %4 = getelementptr inbounds { %Callable*, %Qubit* }, { %Callable*, %Qubit* }* %2, i32 0, i32 1
+  %5 = call %Callable* @__quantum__rt__callable_create([4 x void (%Tuple*, %Tuple*, %Tuple*)*]* @Microsoft__Quantum__Testing__QIR_____GUID___NoArgs, [2 x void (%Tuple*, i64)*]* null, %Tuple* null)
+  store %Callable* %5, %Callable** %3, align 8
+  store %Qubit* %q, %Qubit** %4, align 8
+  call void @Microsoft__Quantum__Simulation__QuantumProcessor__Extensions_____GUID___ApplyIfOne__body(%Result* %0, { %Callable*, %Qubit* }* %2)
```

Comment:
Where did the error occur before this change? Was it the `ApplyIfOne` call?

Reply:
Yes, it was the argument to the call. The issue was that the argument to ApplyIfOne was a tuple that the ClassicalControl transformation generated the AST for. While the data structures of the compiler enforce that there cannot be any tuple *type* that has less than two arguments, it does not enforce the same for the expression. What was hence injected by ClassicalControll was a typed expression that was of type Qubit and of expression kind ValueTuple [Qubit]. The QIR emission then build a tuple but tried to pass it as Qubit*.